Use of NFS mount for remote backup locations added (#877)
MichaIng/DietPi-Docs: docs/dietpi_tools/system_maintenance.md

=== "Settings files"

    Generally, the settings of the DietPi-Backup are changed via the `dietpi-backup` command menu entries.

    The system stores these settings in the files `/boot/dietpi/.dietpi-backup_settings` and `/boot/dietpi/.dietpi-backup_inc_exc`, which are generated from `dietpi-backup` automatically. Therefore, the files do not need to be changed manually by the user.

=== "Backup file selection (Filter)"

    The definition which files are used for the backup procedure is defined via the option "Filter" (see screenshot above). This opens `nano` to edit the include/exclude definitions for the backup.  
    The filter definition syntax is described within the file itself.

    ![DietPi-Backup filter option screenshot](../assets/images/dietpi-backup_filter-option.jpg){: width="681" height="330" loading="lazy"}

    The file containing the filter definitions is `/boot/dietpi/.dietpi-backup_inc_exc`.

=== "Space check"

    A space check on the target location prior to the backup process can be enabled/disabled. This might be an option if there is surely enough disk space available.  

>     Enabling the space check makes the backup a bit more safe, disabling it speeds it up.

Reviewer: ```suggestion
    Enabling the space check makes the backup a bit more safe, disabling it speeds it up, especially when using a network drive.
```

=== "Logging"

    Logging information about the backup procedure is given within the files `.dietpi-backup_stats` and `.dietpi-backup.log` which are located in the backup target directory ("Location" option):

    - `.dietpi-backup_stats` gives a list of completed operations with time and date
    - `.dietpi-backup.log` gives a list of every processed file

### Location of the backup

The location of the backup can be set to any directory which is not part of the backup itself. Especially, these options are typical:

- Backup to/Restore from a connected USB stick or USB disk

> - Backup to/Restore from a large system disk mounted below `/media` (because `/media` is not backed up by default)

Reviewer: ```suggestion
```
This is probably redundant in combination with the first option. For clarification, also about "which is not part of the backup itself": `dietpi-backup` adds the target location automatically to excludes, so one can really chose anything as backup location and a sync loop is ruled out. The only limitation is that the target filesystem needs to support UNIX permissions and symlinks.

- Backup to/Restore from an network drive (Linux file system needed!)

!!! info "Backup to/Restore from a network drive: Only Linux file systems are usable (NFS)"

    DietPi-Backup is purely based on `Rsync`, i.e., DietPi-Backup uses a backup on a file by file basis and is not based on an archive algorithm. This leads to the fact that file permissions/attributes are only preserved correctly in case of a Linux file system as the target directory. In other cases (e.g. a Samba share), file permissions are not saved in the right manner and DietPi-Backup will not work appropriate.

    **Therefore, DietPi-Backup should only be used to/from a directory which handles Linux file attributes like NFS.**

> 
>     Remark: In the case that the `rsync` package is not installed, this is done automatically once you start a backup or restore.

Reviewer: ```suggestion
```
Probably no need to mention this, user will see it, and there is no harm if one installs it manually.

### Automatic daily backup

`Dietpi-Backup` gives the option of an automatic daily backup function (controlled via the Linux `cron` mechanism).

It contains these options (see screenshot above):

- "Daily Backup": Activates the daily backup

> - "Amount": Sets the number of backups to be kept. Backups are rotated automatically, if the maximum amount has been reached, the oldest backup is used as basis for the incremental new backup sync

Reviewer: Actually the `Amount` option is also effective for manual backups.

Author: Changed with the next commits.

    - `.dietpi-backup_stats` gives a list of completed operations with time and date
    - `.dietpi-backup.log` gives a list of every processed file

### Important DietPi-Backup hints

> !!! note "File system type of the backup directory: NFS needed!"

Reviewer: It should be made clear that this applies for network drives only, actually not even there, as SSHFS would work as well. Probably better:
```suggestion
!!! note "File system type of the backup directory: Linux filesystem needed!"
```
or the other way round, that Samba is not supported?

    DietPi-Backup is purely based on `Rsync`, i.e., DietPi-Backup uses a backup on a file by file basis and is not based on an archive algorithm. This leads to the fact that file permissions/attributes are only preserved correctly in case of a Linux file system as the target directory.   
    In other cases (e.g. a Samba share), file permissions are not saved in the right manner and DietPi-Backup will not work appropriate.

>     **Therefore, DietPi-Backup should only be used to/from a directory which handles Linux file attributes (e.g. NFS).**

Reviewer: Same here, NFS is only a good example if the whole section is clearly about network drives. If it is a general section about filesystem requirements, then probably `ext4` is a better example.

Author: Changed the text a bit.
